### PR TITLE
Bigtable: 03. Surface: SampleRowKeys

### PR DESCRIPTION
**Pull request by igorbernstein2**
Adds the surface to wrap SampleRowKeys RPC. The surface:
- hides the fact that the RPC is streaming
- wraps the request as a simple string table id
- wraps the response in KeyOffset

This has been reviewed Kevin. However some things have changed since the original review:
- The original code was adapted to use the StubSettings
- static import of Truth.assertThat

### CODE REVIEW

**Comment by igorbernstein2** [comment]
rebased

**Comment by igorbernstein2** [comment]
PTAL after #2861. This is ready for review

**Comment by pongad** [comment]
@igorbernstein2 Looks like we need to rebase?

**Comment by igorbernstein2** [comment]
rebased

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/KeyOffset.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  /**
+   * Sorted streamed sequence of sample row keys in the table. The table might have contents before
+   * the first row key in the list and after the last one, but a key containing the empty string
+   * indicates "end of table" and will be the last response given, if present.
+   *
+   * <p>Note that row keys in this list may not have ever been written to or read from, and users
+   * should therefore not make any assumptions about the row key structure that are specific to
+   * their use case.
+   */
+  public abstract ByteString geyKey();
```
Comment (pongad) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/KeyOffset.java]: The docs seem to suggest we have multiple keys but we only have one right?

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -247,6 +251,48 @@ public void readRowsAsync(Query query, ResponseObserver<Row> observer) {
     return stub.createReadRowsCallable(rowAdapter);
   }
 
+  /**
+   * Convenience method to asynchronously return a sample of row keys in the table. The returned row
+   * keys will delimit contiguous sections of the table of approximately equal size, which can be
+   * used to break up the data for distributed tasks like mapreduces. The returned callable object
```
Comment (garrettjonesgoogle) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: This method returns an `ApiFuture`, not a callable

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/KeyOffset.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public abstract class KeyOffset {
+  @InternalApi
+  public static KeyOffset create(ByteString key, long offsetBytes) {
+    return new AutoValue_KeyOffset(key, offsetBytes);
+  }
+
+  /**
+   * A row key value that is returned as part of the response of
+   * {@link com.google.cloud.bigtable.data.v2.BigtableDataClient#sampleRowKeys(String)}. The key
+   * represents end boundary of one of the contiguous section in a list of approximately equal sized
```
Comment (garrettjonesgoogle) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/wrappers/KeyOffset.java]: s/section/sections/

**Comment by igorbernstein2** [comment]
Fixed docs. Please merge when ready

**Comment by igorbernstein2** [comment]
Thank you!

**Pull request merged**